explainer-server/app/controllers/ApiController.scala

```diff
@@ -76,6 +76,50 @@ class ExplainerApiImpl(
     explainerToPublish.map(CsAtom.atomToCsAtom)
   }
 
+  override def addTagToExplainer(explainerId: String, tagId: String): Future[CsAtom] = {
+    for {
+      csatom <- explainerDB.load(explainerId).map(CsAtom.atomToCsAtom)
+    }yield{
+      val csexplaineratom: CsExplainerAtom = csatom.data
+      val tags: Option[List[String]] = csexplaineratom.tags
+      val newtags: Option[List[String]] = tags match {
+        case None => Some(List(tagId))
+        case Some(list) => Some( (tagId :: list).distinct.sorted )
+      }
+      val newCsExplainerAtom: CsExplainerAtom = CsExplainerAtom(csexplaineratom.title, csexplaineratom.body, csexplaineratom.displayType, newtags)
+      val newCsAtom: CsAtom = CsAtom(csatom.id, newCsExplainerAtom, csatom.contentChangeDetails)
+      explainerDB.store(CsAtom.csAtomToAtom(newCsAtom))
+      explainerStore.updateLastModified(explainerId, user)
+    }
+    load(explainerId)
+  }
+
+  override def removeTagFromExplainer(explainerId: String, tagId: String): Future[CsAtom] = {
+    for {
+      csatom <- explainerDB.load(explainerId).map(CsAtom.atomToCsAtom)
+    }yield{
+      val csexplaineratom: CsExplainerAtom = csatom.data
+      val tags: Option[List[String]] = csexplaineratom.tags
+      val newtags: Option[List[String]] = tags match {
+        case None => None
+        case Some(list) => {
+          val x = (list diff List(tagId)).sorted
+          // We do the following check to prevent to return an empty list, which causes some problems.
+          if(x.size==0){
+            None
+          }else{
+            Some(x)
+          }
+        }
+      }
+      val newCsExplainerAtom: CsExplainerAtom = CsExplainerAtom(csexplaineratom.title, csexplaineratom.body, csexplaineratom.displayType, newtags)
+      val newCsAtom: CsAtom = CsAtom(csatom.id, newCsExplainerAtom, csatom.contentChangeDetails)
+      explainerDB.store(CsAtom.csAtomToAtom(newCsAtom))
+      explainerStore.updateLastModified(explainerId, user)
```

Comment:
If you decide to go for using `copy` above, you could do the same thing here to avoid this last block of statements